File: 0123-best-time-to-buy-and-sell-stock-iii/0123-best-time-to-buy-and-sell-stock-iii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit1=0,maxProfit2=0;
        int minPrice1=INT_MAX,minPrice2=INT_MAX;
        for(int i=0;i<prices.size();++i)
        {
            minPrice1=min(minPrice1,prices[i]);
            maxProfit1=max(maxProfit1,prices[i]-minPrice1);
            
            minPrice2=min(minPrice2,prices[i]-maxProfit1);
            maxProfit2=max(maxProfit2,prices[i]-minPrice2);
        }
        return maxProfit2;
        
    }
};

File: 0123-best-time-to-buy-and-sell-stock-iii/0123-best-time-to-buy-and-sell-stock-iii.java
class Solution {
    public int maxProfit(int[] prices) {
         int maxProfit1=0,maxProfit2=0;
        int minPrice1=Integer.MAX_VALUE,minPrice2=Integer.MAX_VALUE;
        for(int i=0;i<prices.length;++i)
        {
            minPrice1=Math.min(minPrice1,prices[i]);
            maxProfit1=Math.max(maxProfit1,prices[i]-minPrice1);
            minPrice2=Math.min(minPrice2,prices[i]-maxProfit1);
            maxProfit2=Math.max(maxProfit2,prices[i]-minPrice2);
        }
        return maxProfit2;
    }
}